File: server/board.py
class Board:
    def __init__(self, board_id=None, name=None):
        self.board_id = board_id
        self.color = {"red": 255, "green": 255, "blue": 255}
        self.led_intensity = 50
        self.name = name
        self.has_update = True
        self.auto_adjust_light = False
        self.setpoint = None
        self.blue_light_filter = False


    def __str__(self):
        return str(self.__dict__)

    
    def board_hash(self):
        return \
            self.color["red"]*17 + \
            self.color["green"]*31 + \
            self.color["blue"]*43 + \
            self.led_intensity*59


    def __eq__(self, other):
        """Overrides the default implementation"""
        if isinstance(other, Board):
            return self.board == other.board_id
        return False


    '''
    def to_dict(self):
        md = dict()
        
        md["board_id"] = self.board_id
        md["color"] = self.color
        md["led_intensity"] = self.led_intensity
        md["name"] = self.name
        md["has_update"]
    '''

File: android/IoTApplication/app/src/main/java/com/example/iotapplication/Board.java
package com.example.iotapplication;

import com.google.gson.annotations.SerializedName;

public class Board {
    private String name;
    @SerializedName("board_id")
    private String boardId;
    private boolean hasUpdate;
    @SerializedName("led_intensity")
    private int ledIntensity;
    @SerializedName("auto_adjust_light")
    private boolean autoAdjustLight;
    private int setpoint;
    private RGBModel color;
    @SerializedName("blue_light_filter")
    private boolean blueLightFilter;

    public Board(String boardId, boolean hasUpdate, int ledIntensity, boolean autoAdjustLight, int setpoint, RGBModel color, boolean blueLightFilter) {
        this.boardId = boardId;
        this.hasUpdate = hasUpdate;
        this.ledIntensity = ledIntensity;
        this.autoAdjustLight = autoAdjustLight;
        this.setpoint = setpoint;
        this.color = color;
        this.blueLightFilter = blueLightFilter;
    }

    public String getBoardId() {
        return boardId;
    }

    public void setBoardId(String boardId) {
        this.boardId = boardId;
    }

    public boolean isHasUpdate() {
        return hasUpdate;
    }

    public void setHasUpdate(boolean hasUpdate) {
        this.hasUpdate = hasUpdate;
    }

    public int getLedIntensity() {
        return ledIntensity;
    }

    public void setLedIntensity(int ledIntensity) {
        this.ledIntensity = ledIntensity;
    }

    public boolean isAutoAdjustLight() {
        return autoAdjustLight;
    }

    public void setAutoAdjustLight(boolean autoAdjustLight) {
        this.autoAdjustLight = autoAdjustLight;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSetpoint() {
        return setpoint;
    }

    public void setSetpoint(int setpoint) {
        this.setpoint = setpoint;
    }

    @Override
    public String toString() {
        return "Board{" +
                "name='" + name + '\'' +
                ", boardId='" + boardId + '\'' +
                ", hasUpdate=" + hasUpdate +
                ", ledIntensity=" + ledIntensity +
                ", autoAdjustLight=" + autoAdjustLight +
                ", setpoint=" + setpoint +
                ", color=" + color +
                ", blueLight=" + blueLightFilter +
                '}';
    }

    public RGBModel getColor() {
        return color;
    }

    public void setColor(RGBModel color) {
        this.color = color;
    }

    public boolean isBlueLightFilter() {
        return blueLightFilter;
    }

    public void setBlueLightFilter(boolean blueLightFilter) {
        this.blueLightFilter = blueLightFilter;
    }
}

